cloud/ec2_client.go

```diff
@@ -97,15 +100,17 @@ const (
 )
 
 // Create a new aws-sdk-client if one does not exist, otherwise no-op.
-func (c *awsClientImpl) Create(creds *credentials.Credentials, region string) error {
+func (c *awsClientImpl) create(creds *credentials.Credentials, region string, forceful bool) error {
```

Comment:
I'd refactor this to avoid passing a bool.

Reply:
Do you mean into 2 functions, or something else?

Reply:
Yes two functions.

Reply:
@tychoish There are already two functions, `Create` and `CreateForcefully`. They call `create`, a private function.

Reply:
I think the internal API should also have two functions, or rather than doing the foreceful thing, you store the credentials pointer with the cached session, and don't need "foreceful" reset, just reset if the credentials don't match the cached version.